airflow/jobs.py

```diff
@@ -2617,25 +2638,11 @@ def heartbeat_callback(self, session=None):
                 self.log.warning("Recorded pid {ti.pid} does not match the current pid "
                                 "{current_pid}".format(**locals()))
                 raise AirflowException("PID of job runner does not match")
-        elif self.task_runner.return_code() is None and hasattr(self.task_runner, 'process'):
-            if ti.state in [State.NONE, State.SCHEDULED, State.QUEUED]:
-                self.log.warning(
-                    "State of task %s has been externally set to %s. I am still running this task, "
-                    "so I'm setting the state back to RUNNING.",
-                    ti.task_id,
-                    ti.state
-                )
-                ti.state = State.RUNNING
-                session.merge(ti)
-                session.commit()
-                return
-            else:
-                self.log.warning(
-                    "State of this instance has been externally set to %s. Taking the poison pill.",
-                    ti.state
-                )
-                self.task_runner.terminate()
-                self.terminating = True
-
-        session.merge(ti)
-        session.commit()
```

Comment:
This is reverting changes to accommodate side effects of the old orphan adoption process.